### PR TITLE
Flask login returns email as _id instead of risky ObjectId id

### LINKED ISSUE #1467: Login crashes for users that have _id = Objectid

### PR DESCRIPTION
**Pull request by northwestwitch**
Fix #1467

**How to test**:
1. Create a new user from mongoDB compass. Don't provide any custom _id so it will be of type ObjectID. Example:
image

1. Try to login with this user from master branch and this branch. 
1. Test even from Scout stage trying to log in a user that has ObjectID as _id 

**Expected outcome**:
Login from master should give error while from this branch it should work.
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by dNil
- [x] tests executed by dNil

### CODE REVIEW

**Comment by dnil** [comment]
Yes, not a bad id!

File: scout/server/blueprints/login/models.py [hunk 1]
```diff
@@ -11,7 +11,7 @@ def __init__(self, user_data):
             setattr(self, key, value)
 
     def get_id(self):
-        return self._id
+        return self.email
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/server/blueprints/login/models.py]: Have you tested it? It should work, right?

[the remark this answers is not in the document]
Comment (dnil) [reply scout/server/blueprints/login/models.py]: I'm just musing from the sofa; no code has been executed during this.. I can give it a spin.

[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/server/blueprints/login/models.py]: Nope, it doesn't login. At least on a local installation.

[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/server/blueprints/login/models.py]: It works if id == email, not if it's an objectID. 

I think it's safe to return email. This is how it was before I changed it in the LDAP PR : https://github.com/Clinical-Genomics/scout/pull/1401/files#diff-d8fca6f47dbde57509fec2cb29be7a46L14

**Comment by dnil** [comment]
So for starters, there is a high degree of weirdness already with the master solution:
Screenshot 2019-11-09 at 11 14 48
Screenshot 2019-11-09 at 11 16 53
Screenshot 2019-11-09 at 11 17 08
Now, onto the patch..

**Comment by northwestwitch** [comment]
> So for starters, there is a high degree of weirdness already with the master solution:

It's OK Clark Kent is superman! XD

**Comment by dnil** [comment]
Your version tests just fine locally:
Screenshot 2019-11-09 at 11 29 59
Screenshot 2019-11-09 at 11 30 12
Screenshot 2019-11-09 at 11 31 48
Screenshot 2019-11-09 at 11 31 38

File: scout/server/blueprints/login/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dnil) [reply scout/server/blueprints/login/models.py]: Concur, my version won't work without a few lines in the user adapter as well to conditionally search for ObjectId(str) if the str is not an email.

**Comment by northwestwitch** [comment]
> But I am not sure we can predict whether Clark will wear a suit or not after hitting login


> k will wear a suit or not after hitting login.

That's why we should do the winter cleaning in that database and fix all counts so there is max an account per user!

**Comment by dnil** [comment]
So the remaining concern would be what happens to users that are already logged in to either when we switch. But as far as my caffeine deprived cortex reports that should land them on one of the two options, without crashes, right?

**Comment by northwestwitch** [comment]
> So the remaining concern would be what happens to users that are already logged in to either when we switch. But as far as my caffeine deprived cortex reports that should land them on one of the two options, without crashes, right?

I guess so. This is how things worked before the LDAP PR anyway, and nobody complained!

**Comment by dnil** [comment]
Is it on stage already?

**Comment by dnil** [comment]
I can report error free login to stage from my chrome session of yesterday. And I have the power again! ✊ That settles it, this merges.

**Comment by northwestwitch** [comment]
> Is it on stage already?

No. I don't have my old phone around for the 2 thing authentication

**Comment by northwestwitch** [comment]
Wait, let's revert the test to check if it's the email!

**Comment by northwestwitch** [comment]
I've re-checked the LDAP login on my local computer -- > It works!

**Comment by dnil** [comment]
Lets revisit this again after/when we do the winter cleaning. Separating email and id is the long term way to do it - it is not really a very transparent thing explain to other sites as it stands now.

**Comment by henrikstranneheim** [comment]
Great work guys!

**Pull request merged**